### ISSUE TITLE
Discrepancy in lazy translations

### ISSUE THREAD

**Issue by stovmascript** [lingui/js-lingui#1432]
**Describe the bug**
The documentation about [lazy translations](https://lingui.dev/tutorials/react-patterns#lazy-translations) shows how you can use `defineMessage` and pass it to either `Trans` or `i18n._()`. When I use `i18n._()`, extracting messages prints out `Missing message ID, skipping.` warnings.

**To Reproduce**
You can use the code provided in the documentation and run `lingui extract`:

```js
// src/I18nTest.js
import { i18n } from '@lingui/core'
import { defineMessage, Trans } from '@lingui/macro'

const favoriteColors = [
  defineMessage({ message: 'Red' }),
  defineMessage({ message: 'Orange' }),
  defineMessage({ message: 'Yellow' }),
  defineMessage({ message: 'Green' }),
]

export default function ColorList() {
  return (
    <ul>
      {favoriteColors.map((color, index) => (
        <li key={index}>
          <Trans id={color.id} />
        </li>
      ))}
    </ul>
  )
}

export function getTranslatedColorNames() {
  return favoriteColors.map((color) => i18n._(color))
}
```

**Expected behavior**
Both methods should work the same, without printing warnings.

**Additional context**

- jsLingui version ``3.17.1`` (`yarn lingui --version` prints "dev"?)
- Babel version ``7.20.12``
- Your Babel config (e.g. `.babelrc`) or framework you use (Create React App, Meteor, etc.)
  - using `"@lingui/swc-plugin": "^0.2.0"`
  - with Next.js in a JS codebase
    ```js
    // lingui.config.js
    extractBabelOptions: {
      presets: ['@babel/preset-react'],
    },

    // next.config.js
    experimental: {
      swcPlugins: [
        [
          '@lingui/swc-plugin',
          {
            // the same options as in .swcrc
          },
        ],
      ],
    },
    ```

**Comment by thekip** [comment]
Yes, that's a bug. Unfortunately there are no obvious solution, because either it's break backward compatibility with existing code or will disable this validation at all, which is also kinda breaking change. 

I will definitely consider this issue in v4.  Thanks for reporting.

As for now, as workaround you can rename i18n object to something and pass descriptor into it. The extractor matching only by identifier name, so this would work

```ts
const _i18n = i18n;

export function getTranslatedColorNames() {
  return favoriteColors.map((color) => _i18n._(color))
}
```

**Comment by thekip** [comment]
Just checked this on the `next` branch, and it seems the issue got fixed by #1358

I even added test, which i don't remember 😅
